vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -65,10 +91,12 @@ class CostMap {
         }
     }
 
+    // Map node id to segment id.
```

Comment:
Why is this cache needed, and why is it part of the cost map?

Reply:
It lowers the number of cache hits per lookup, it was a clear signal during the initial connection box optimization pass